### DISCUSSION TITLE
Completion does not trigger while type or tab?

### DISCUSSION THREAD

**Discussion by tmpm697** [hrsh7th/nvim-cmp#1686] [General]
I have this block in setup():
```
		mapping = {
			["<C-n>"] = cmp.mapping.select_next_item({ behavior = cmp.SelectBehavior.Insert }),
			["<C-p>"] = cmp.mapping.select_prev_item({ behavior = cmp.SelectBehavior.Insert }),
			["<Tab>"] = cmp.mapping.select_next_item({ behavior = cmp.SelectBehavior.Insert }),
			["<S-Tab>"] = cmp.mapping.select_prev_item({ behavior = cmp.SelectBehavior.Insert }),
			["<C-Space>"] = cmp.mapping.complete(),
			["<CR>"] = cmp.mapping.confirm({ select = true }),
		},

```

that expect to trigger cmp to display candidates while typing or press `tab` to pop-up candiates but does not happens in my case.

Is there smth wrong with above setup with recent update from cmp?

**Comment by hedyhli** [comment]
Does <kbd>C-space</kbd> display candidates for you?

> Completion does not trigger while type

It should trigger automatically if you did not set autocomplete to false.

Otherwise your C-space mapping should trigger completion.

> Is there smth wrong with above setup

The tab bindings look correct but it will only work if there's already a menu

If you want to have TAB for selecting next candidate and also open the menu if it's not already opened, you have to use a custom function, see [this wiki page](https://github.com/hrsh7th/nvim-cmp/wiki/Example-mappings#super-tab-like-mapping). You can remove the snippet checks if you're not using a snippet engine:

```lua
local has_words_before = function()
  unpack = unpack or table.unpack
  local line, col = unpack(vim.api.nvim_win_get_cursor(0))
  return col ~= 0 and vim.api.nvim_buf_get_lines(0, line - 1, line, true)[1]:sub(col, col):match("%s") == nil
end

-- ... mappings table ...
["<Tab>"] = cmp.mapping(function(fallback)
      if cmp.visible() then
        cmp.select_next_item()
      elseif has_words_before() then
        cmp.complete()
      else
        fallback() -- The fallback function sends a already mapped key. In this case, it's probably `<Tab>`.
      end
    end, { "i" }),
-- ... mappings table ...
```
